src/check/parse/tokenize.zig

```diff
@@ -637,7 +670,97 @@ pub const Cursor = struct {
+                    const is_trivia = @select(bool, is_space, true_vec, @select(bool, is_tab, true_vec, @select(bool, is_newline, true_vec, @select(bool, is_cr, true_vec, false_vec))));
+                    const all_trivia = @reduce(.And, is_trivia);
+
+                    if (all_trivia) {
+                        // Check if there are any newlines or carriage returns in the chunk
+                        const has_line_ending = @reduce(.Or, @select(bool, is_newline, is_newline, is_cr));
+
+                        if (has_line_ending) {
+                            // Process chunk with newlines byte by byte to handle them correctly
+                            // This is still faster than pure scalar processing
+                            var i: usize = 0;
+                            while (i < SIMD_WIDTH and self.pos < self.buf.len) : (i += 1) {
+                                const b = self.buf[self.pos];
+                                if (b == ' ') {
+                                    self.pos += 1;
+                                    if (sawNewline) indent += 1;
+                                } else if (b == '\t') {
+                                    self.pos += 1;
+                                    if (sawNewline) {
+                                        indent = (indent + self.tab_width) & ~(self.tab_width - 1);
+                                    }
+                                } else if (b == '\n') {
+                                    self.pos += 1;
+                                    sawNewline = true;
+                                    indent = 0;
+                                } else if (b == '\r') {
+                                    self.pos += 1;
+                                    sawNewline = true;
+                                    indent = 0;
+                                    // Handle CRLF
+                                    if (self.pos < self.buf.len and self.buf[self.pos] == '\n') {
+                                        self.pos += 1;
+                                        i += 1; // Account for the extra character consumed
+                                    } else {
+                                        self.pushMessageHere(.MisplacedCarriageReturn);
+                                    }
+                                }
+                            }
+                            self.chomp_trivia_simd_chunks_processed += 1;
+                            continue;
+                        } else {
+                            // No newlines, process the entire chunk with SIMD
+                            const ones: @Vector(SIMD_WIDTH, u16) = @splat(1);
+                            const zeros: @Vector(SIMD_WIDTH, u16) = @splat(0);
+
+                            // Use SIMD to count spaces and tabs efficiently
+                            const space_count = @reduce(.Add, @select(u16, is_space, ones, zeros));
```

Comment:
I don't get why these selects are needed. Isn't `is_space` a boolean vector? Can't you just cast it to integer and reduce?